soapboxd/application.go

```diff
@@ -344,3 +344,50 @@ func (s *server) GetApplication(ctx context.Context, req *pb.GetApplicationReque
 func setPbTimestamp(ts *gpb.Timestamp, t time.Time) {
 	ts.Seconds = t.Unix()
 }
+
+func (s *server) DeleteApplication(ctx context.Context, app *pb.Application) (*pb.Empty, error) {
+	// start a terraform job in the background
+	go s.deleteAppInfrastructure(app)
+
+	tx, err := s.db.Begin()
+	if err != nil {
+		return nil, errors.Wrap(err, "beginning transaction")
+	}
+
+	query := `DELETE FROM applications WHERE id = $1`
```

Comment:
I think we should probably never actually delete rows from the applications table, but mark them as deleted (i.e., add a `deleted` boolean column). Couple of reasons, one is that it would break the new activity feed, which needs the app id to look up an application's info (like name, etc.). (We'd probably also need a `deleted_at` timestamp column.) Also I think it's good to have a de facto record of all apps created in this way, and a user is only ever going to create at max on the order of tens of applications, so the table won't ever get too large (even if they did on order of hundreds of apps).

Reply:
Also regarding updating the db, it should only be done once the application's resources are finished being destroyed by Terraform, because if for some reason `terraform destroy` fails and needs to be re-run, the pointer to the application will be gone from the user if the db is updated first.

Reply:
Oh, and also, regarding the activity feed, please add a new activity type where appropriate and make sure that delete application activity is recorded. (You'll need to merge in master.)